In [1]:
import cv2
import pytesseract

In [2]:
pytesseract.pytesseract.tesseract_cmd = 'C:\\Program Files\\Tesseract-OCR\\tesseract'

In [3]:
img = cv2.imread('4.PNG')
img = cv2.cvtColor(img,cv2.COLOR_BGR2RGB)
cv2.imshow('Image', img)
cv2.waitKey(0)

-1

In [4]:
text = pytesseract.image_to_string(img, lang='eng')

In [5]:
text

'HELLO\nWORLD\n'

- image_to_boxes returns the values of [character-xpoint-ypoint-width-height]

In [6]:
boxes= pytesseract.image_to_boxes(img)
print(boxes)

H 153 121 167 141 0
E 172 121 184 141 0
L 188 121 199 141 0
L 202 121 212 141 0
O 213 121 231 141 0
W 143 68 169 88 0
O 171 68 189 88 0
R 193 68 207 88 0
L 211 68 221 88 0
D 224 68 240 88 0



In [8]:
#the height and width of img
hImg, wImg, _ = img.shape
print(hImg,wImg, _)

189 381 3


### cv2.rectangle

#### cv2.rectangle(image_name,start point as a tuple, end point as a tuple,color,thickness)

Both OpenCV and Pytesseract use the same coordinate system, where the origin (0,0) is located at the top-left corner of the image. The X-axis increases from left to right, and the Y-axis increases from top to bottom.

Now we gonna try to figure out how start and end points work,i became obsessed with how start and end points are perceived.
start point:(x1,yı)
end point:(x2,y2)

- start>>> (x,hImg-y)
- end>>> (w, hImg-h)


Just focus on H character.
H 153 121 167 141 0

start>>>(153,189-121)
end>>>>>(167,189-141)

In [12]:
cv2.rectangle(img,(153,68),(167,48),(255,255,0),1)
cv2.imshow('Try', img)
cv2.waitKey(0)
#output correctly bounds the H character with blue lines

-1

When I searched some pages, I read that x1,y1 is the upper left corner and x2,y2 is the bottom right corner. But this idea doesn't work when you visualize the coordinate system. Because....

In [14]:
cv2.rectangle(img,(167,48),(153,68),(255,0,255),1)
cv2.imshow('Try', img)
cv2.waitKey(0)
#this output also correctly bounds the H character with red(seems pink) line

-1

In [17]:
cv2.rectangle(img,(153,48),(167,68),(0,255,255),1)
cv2.imshow('Try', img)
cv2.waitKey(0)
#this output also correctly bounds the H character with yellow line

-1

In [18]:
cv2.rectangle(img,(167,68),(153,48),(255,255,0),1)
cv2.imshow('Try', img)
cv2.waitKey(0)
#this output also correctly bounds the H character with blue line.

-1

## So I think it doesn't matter what order you write the points x1,x2 and y1,y2 for the cv2.rectangle function. Just give it a two tuples as (x,y). It gonna fıgure out which one is left-top corner, right-bottom corner